core-it/src/test/kotlin/org/evomaster/core/problem/rest/service/resource/ResourceTestBase.kt

```diff
@@ -306,45 +306,44 @@ abstract class ResourceTestBase : ExtractTestBaseH2(), ResourceBasedTestInterfac
         assertNotNull(individual)
         assert(individual!!.getResourceCalls().size == 1)
         val addSpec = MutatedGeneSpecification()
-        structureMutator.mutateRestResourceCalls(individual, RestResourceStructureMutator.MutationType.ADD, addSpec)
-        assert(addSpec.mutatedPosition.size == 1)
-        assert(addSpec.addedGenes.isNotEmpty() || individual.getResourceCalls()[addSpec.mutatedPosition.first()].actions.flatMap { it.seeGenes() }.isEmpty())
+        val evaluatedIndividual = EvaluatedIndividual(FitnessValue(0.0), individual, listOf())
+        structureMutator.mutateRestResourceCalls(individual,  RestResourceStructureMutator.MutationType.ADD, addSpec)
+        assert(addSpec.mutatedGenes.mapNotNull { it.resourcePosition }.toSet().size == 1)
+        assert(addSpec.getAdded(true).isNotEmpty())
         assert(individual.getResourceCalls().size == 2)
 
         val first = individual.getResourceCalls()[0].getResourceNode()
         val second = individual.getResourceCalls()[1].getResourceNode()
         val swapSpec = MutatedGeneSpecification()
-        structureMutator.mutateRestResourceCalls(individual, RestResourceStructureMutator.MutationType.SWAP, swapSpec)
-        assert(swapSpec.addedGenes.isEmpty())
-        assert(swapSpec.removedGene.isEmpty())
-        assert(swapSpec.mutatedPosition.size == 2)
+        structureMutator.mutateRestResourceCalls(individual,  RestResourceStructureMutator.MutationType.SWAP, swapSpec)
+        assert(swapSpec.getSwap().isNotEmpty())
 
         assert(individual.getResourceCalls()[1].getResourceNode().getName() == first.getName())
         assert(individual.getResourceCalls()[0].getResourceNode().getName() == second.getName())
 
         val previousIndividual = individual.copy() as RestIndividual
         val delSpec = MutatedGeneSpecification()
-        structureMutator.mutateRestResourceCalls(individual, RestResourceStructureMutator.MutationType.DELETE, delSpec)
-        assert(delSpec.mutatedPosition.size == 1)
-        assert(delSpec.removedGene.isNotEmpty() || previousIndividual.getResourceCalls()[delSpec.mutatedPosition.first()].actions.flatMap { it.seeGenes() }.isEmpty())
+        structureMutator.mutateRestResourceCalls(individual,  RestResourceStructureMutator.MutationType.DELETE, delSpec)
+        assert(delSpec.getRemoved(true).mapNotNull { it.resourcePosition }.toSet().size == 1)
```

Comment:
all these assertions seem expensive, as operations like `mapNotNull` and `toSet` creates new collections. these should be `Lazy` evaluated

Reply:
now it is replaced by `distinctBy`

Reply:
hi. sorry, there was a misunderstanding... :)
i didn't see it was in `test` folder.
`assert` is to define invariants in the SUT (eg, under `main` folder).
in tests, use the `assert*` from JUnit, eg, `assertEquals`, and not the function `assert` from Kotlin.
when invariant are expensive, to do not slow down the SUT, we use `Lazy.assert`, which is not needed in the tests